src/SSHDebugPS/Docker/DockerHelper.cs

```diff
@@ -46,33 +55,55 @@ public static IEnumerable<IContainerInstance> GetLocalDockerContainers()
                         if (args.Trim()[0] != '{')
                         {
                             // output isn't json, command Error
-                            string errorMessage = string.Format(CultureInfo.CurrentCulture, UIResources.CommandExecutionErrorFormat, dockerCommand, args);
-                            throw new CommandFailedException(errorMessage);
+                            errorSB.AppendLine(args);
+                        }
+                        else
+                        {
+                            var containerInstance = DockerContainerInstance.Create(args);
+                            if (containerInstance != null)
+                                containers.Add(containerInstance);
                         }
-
-                        var containerInstance = DockerContainerInstance.Create(args);
-                        if (containerInstance != null)
-                            containers.Add(containerInstance);
                     }
                 });
 
-                commandRunner.Run();
-                // TODO: Switch to cancellable wait
-                resetEvent.WaitOne();
+                commandRunner.Start();
 
-                // might need to throw an exception here too??
-                if (exitCode != 0)
+                bool cancellationRequested = false;
+                VS.VSOperationWaiter.Wait(UIResources.QueryingForContainersMessage, false, (cancellationToken) =>
                 {
-                    Debug.Fail(FormattableString.Invariant($"Exit Code: {exitCode}"));
-                    return null;
+                    while (!resetEvent.WaitOne(2000) && !cancellationToken.IsCancellationRequested)
+                    { }
+                    cancellationRequested = cancellationToken.IsCancellationRequested;
```

Comment:
>cancellationRequested = cancellationToken.IsCancellationRequested; [](start = 20, length = 66)

I don't understand, why are you checking this inside the lambda?

Reply:
Pierson explained it.

---
In reply to: [297897088](https://github.com/microsoft/MIEngine/pull/884#discussion_r297897088) [](ancestors = 297897088)